src/executionFiles/bridges/hop.ts

```diff
@@ -3,117 +3,101 @@ import {
+import axios from 'axios'
 import BigNumber from 'bignumber.js'
-import { ethers, Signer } from 'ethers'
+import { ethers } from 'ethers'
 
-import {
-  ChainId,
-  ChainKey,
-  CoinKey,
-  getChainByKey,
-  ParsedReceipt,
-} from '../../types'
+import { ChainId, ParsedReceipt } from '../../types'
+import { loadTransaction, repeatUntilDone } from '../../utils'
 import { defaultReceiptParsing } from '../utils'
 
-let hop: Hop | undefined = undefined
-
-let bridges: { [k: string]: HopBridge } = {}
-const hopChains: { [k: number]: Chain } = {
-  [getChainByKey(ChainKey.ETH).id]: Chain.Ethereum,
-  [getChainByKey(ChainKey.POL).id]: Chain.Polygon,
-  [getChainByKey(ChainKey.DAI).id]: Chain.xDai,
-  [getChainByKey(ChainKey.OPT).id]: Chain.Optimism,
-  [getChainByKey(ChainKey.ARB).id]: Chain.Arbitrum,
+const hopChains: { [k: number]: string } = {
+  [ChainId.ETH]: 'ethereum',
+  [ChainId.POL]: 'polygon',
+  [ChainId.DAI]: 'xdai',
+  [ChainId.OPT]: 'optimism',
+  [ChainId.ARB]: 'arbitrum',
 
   // Testnet; Hop SDK changes the underlying id of these chains according to the instance network
   //network 'goerli'
-  [getChainByKey(ChainKey.GOR).id]: Chain.Ethereum,
-  [getChainByKey(ChainKey.MUM).id]: Chain.Polygon,
+  [ChainId.GOR]: 'ethereum',
+  [ChainId.MUM]: 'polygon',
 }
 
-const supportedTestnetChains: number[] = [ChainId.GOR, ChainId.MUM]
+const getSubgraphUrl = (chainId: ChainId): string => {
+  const chain = hopChains[chainId]
 
-// get these from https://github.com/hop-protocol/hop/blob/develop/packages/sdk/src/models/Token.ts
-const hopTokens: { [k: string]: string } = {
-  USDC: Token.USDC,
-  USDT: Token.USDT,
-  MATIC: Token.MATIC,
-  DAI: Token.DAI,
-  ETH: Token.ETH,
-}
-const isInitialized = () => {
-  if (hop === undefined)
-    throw TypeError('Hop instance is undefined! Please initialize Hop')
-}
-const init = (signer: Signer, chainId: number, toChainId: number) => {
-  const isChainTest = supportedTestnetChains.includes(chainId) ? true : false
-  const isToChainTest = supportedTestnetChains.includes(toChainId)
-    ? true
-    : false
-  // goerli <-> mumbai
-  if (isChainTest && isToChainTest) {
-    hop = new Hop('goerli')
-  } else {
-    hop = new Hop('mainnet')
-  }
-  bridges = {
-    USDT: hop.connect(signer).bridge('USDT'),
-    USDC: hop.connect(signer).bridge('USDC'),
-    MATIC: hop.connect(signer).bridge('MATIC'),
-    DAI: hop.connect(signer).bridge('DAI'),
-    ETH: hop.connect(signer).bridge('ETH'),
+  if (!chain) {
+    throw new Error(`Unsupported chainId passed: ${chainId}`)
   }
+
+  return `https://api.thegraph.com/subgraphs/name/hop-protocol/hop-${chain}`
 }
 
-const getHopBridge = (bridgeCoin: CoinKey) => {
-  isInitialized()
-  if (!Object.keys(bridges).length) {
-    throw Error(
-      'No HopBridge available! Initialize Hop implementation first via init(signer: JsonRpcSigner, chainId: number, toChainId: number)'
-    )
-  }
-  return bridges[bridgeCoin]
+const getTransferIdByTxHash = async (
```

Comment:
Perhaps the function name show contain that we only search for transfers on the sending side.

Reply:
The function allows to search for transfers on both sides, depending on which chain you pass, that's why I gave it a general name. 
But one could argue that it doesn't make so much sense to look on the receiving side 😅  Ill give both functions a similar naming